pkg/scheduler/algorithm/predicates/metadata.go

```diff
@@ -91,21 +101,26 @@ type predicateMetadata struct {
 }
 
 // Ensure that predicateMetadata implements algorithm.PredicateMetadata.
-var _ algorithm.PredicateMetadata = &predicateMetadata{}
+var _ PredicateMetadata = &predicateMetadata{}
 
 // PredicateMetadataProducer function produces predicate metadata.
-type PredicateMetadataProducer func(pm *predicateMetadata)
+type predicateMetadataProducer func(pm *predicateMetadata)
```

Comment:
This one confuses me every time I read it. It's not really in scope of this PR, but could you add a note to the comment that this type is stored in a global variable below and used to modify the return values of `PredicateMetadataProducer`?

Reply:
Done thanks